src/czbenchmarks/tasks/task.py

```diff
@@ -21,6 +29,294 @@ class TaskOutput(BaseModel):
     model_config = {"arbitrary_types_allowed": True}
 
 
+class TaskParameter(BaseModel):
+    """Schema for a single, discoverable parameter."""
+
+    type: str
+    default: Any = None
+    required: bool
+
+
+class TaskInfo(BaseModel):
+    """Schema for all discoverable information about a single benchmark task."""
+
+    name: str
+    display_name: str
+    description: str
+    task_params: Dict[str, TaskParameter]
+    baseline_params: Dict[str, TaskParameter]
+    metrics: List[str]
+
+
+class TaskRegistry:
+    """A registry that is populated automatically as Task subclasses are defined."""
+
+    def __init__(self):
+        self._registry: Dict[str, Type["Task"]] = {}
+        self._info: Dict[str, TaskInfo] = {}
+
+    def register_task(self, task_class: Type["Task"]):
+        """Registers a task class and introspects it to gather metadata."""
+        if inspect.isabstract(task_class) or not hasattr(task_class, "display_name"):
+            return
+
+        key = (
+            getattr(task_class, "display_name", task_class.__name__)
+            .lower()
+            .replace(" ", "_")
+        )
+        self._registry[key] = task_class
+        self._info[key] = self._introspect_task(task_class)
+
+    def _introspect_task(self, task_class: Type["Task"]) -> TaskInfo:
+        """Extracts parameter and metric information from a task class."""
+        try:
+            # 1. Get Task Parameters from the associated Pydantic input model
+            task_params = {}
+            if hasattr(task_class, "input_model") and issubclass(
+                task_class.input_model, BaseModel
+            ):
+                for (
+                    field_name,
+                    field_info,
+                ) in task_class.input_model.model_fields.items():
+                    # Enhanced type information extraction
+                    type_info = self._extract_type_info(
+                        field_info.annotation, field_name
+                    )
+                    task_params[field_name] = TaskParameter(
+                        type=type_info,
+                        default=field_info.default
+                        if field_info.default is not PydanticUndefined
+                        else None,
+                        required=field_info.is_required(),
+                    )
+
+            # 2. Get Baseline Parameters from the compute_baseline method signature
+            baseline_params = {}
+            try:
+                sig = inspect.signature(task_class.compute_baseline)
+                for param in list(sig.parameters.values())[1:]:
```

Comment:
Is the `[1:]` to skip `self`? I'd probably just have an `if param.name == "self": continue` so it's clearer to the reader. Or at least have a comment